azurerm/internal/services/cosmos/resource_arm_cosmosdb_sql_container.go

```diff
@@ -122,12 +123,11 @@ func resourceArmCosmosDbSQLContainerCreate(d *schema.ResourceData, meta interfac
 				return fmt.Errorf("Error checking for presence of creating Cosmos SQL Container %s (Account: %s, Database:%s): %+v", name, account, database, err)
 			}
 		} else {
-			id, err := azure.CosmosGetIDFromResponse(existing.Response)
-			if err != nil {
-				return fmt.Errorf("Error generating import ID for Cosmos SQL Container '%s' (Account: %s, Database:%s)", name, account, database)
+			if existing.ID != nil && *existing.ID != "" {
```

Comment:
we want to raise an error if there's already an existing resource (e.g. an ID is returned) so this wants to become:

```suggestion
			if existing.ID == nil || *existing.ID == "" {
```